spark3/src/main/java/org/apache/iceberg/spark/source/SparkWrite.java

```diff
@@ -260,16 +265,29 @@ public void commit(WriterCommitMessage[] messages) {
   }
 
   private class DynamicOverwrite extends BaseBatchWrite {
+
+    private final IsolationLevel isolationLevel;
+    private final long startingSnapshotId;
+
+    private DynamicOverwrite(IsolationLevel isolationLevel, long startingSnapshotId) {
+      this.isolationLevel = isolationLevel;
+      this.startingSnapshotId = startingSnapshotId;
+    }
+
     @Override
     public void commit(WriterCommitMessage[] messages) {
       ReplacePartitions dynamicOverwrite = table.newReplacePartitions();
+      dynamicOverwrite.validateFromSnapshot(startingSnapshotId);
 
       int numFiles = 0;
       for (DataFile file : files(messages)) {
         numFiles += 1;
         dynamicOverwrite.addFile(file);
       }
 
+      if (isolationLevel == SERIALIZABLE) {
+        dynamicOverwrite.validateNoConflictingAppends();
+      }
```

Comment:
Nit: please add a newline between control flow blocks and the next statement.

Reply:
Sure, done.